builtin/providers/librato/resource_librato_metric.go

```diff
@@ -0,0 +1,432 @@
+						},
+						"aggregate": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceLibratoMetricCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*librato.Client)
+
+	metric := new(librato.Metric)
+	if a, ok := d.GetOk("name"); ok {
+		metric.Name = librato.String(a.(string))
+	}
+	if a, ok := d.GetOk("display_name"); ok {
+		metric.DisplayName = librato.String(a.(string))
+	}
+	if a, ok := d.GetOk("type"); ok {
+		metric.Type = librato.String(a.(string))
+	}
+	if a, ok := d.GetOk("description"); ok {
+		metric.Description = librato.String(a.(string))
+	}
+	if a, ok := d.GetOk("period"); ok {
+		metric.Period = librato.Uint(a.(uint))
+	}
+	if a, ok := d.GetOk("composite"); ok {
+		metric.Composite = librato.String(a.(string))
+	}
+
+	if a, ok := d.GetOk("attributes"); ok {
+
+		attributeData := a.([]interface{})
+		if len(attributeData) > 1 {
+			return fmt.Errorf("Only one set of attributes per metric is supported")
+		}
+
+		if len(attributeData) == 1 && attributeData[0] == nil {
+			return fmt.Errorf("No attributes found in attributes block")
+		}
+
+		attributeDataMap := attributeData[0].(map[string]interface{})
+		attributes := new(librato.MetricAttributes)
+
+		if v, ok := attributeDataMap["color"].(string); ok && v != "" {
+			attributes.Color = librato.String(v)
+		}
+		if v, ok := attributeDataMap["display_max"].(string); ok && v != "" {
+			attributes.DisplayMax = librato.String(v)
+		}
+		if v, ok := attributeDataMap["display_min"].(string); ok && v != "" {
+			attributes.DisplayMin = librato.String(v)
+		}
+		if v, ok := attributeDataMap["display_units_long"].(string); ok && v != "" {
+			attributes.DisplayUnitsLong = *librato.String(v)
+		}
+		if v, ok := attributeDataMap["display_units_short"].(string); ok && v != "" {
+			attributes.DisplayUnitsShort = *librato.String(v)
+		}
+		if v, ok := attributeDataMap["created_by_ua"].(string); ok && v != "" {
+			attributes.CreatedByUA = *librato.String(v)
+		}
+		if v, ok := attributeDataMap["display_stacked"].(bool); ok {
+			attributes.DisplayStacked = *librato.Bool(v)
+		}
+		if v, ok := attributeDataMap["gap_detection"].(bool); ok {
+			attributes.GapDetection = *librato.Bool(v)
+		}
+		if v, ok := attributeDataMap["aggregate"].(bool); ok {
+			attributes.Aggregate = *librato.Bool(v)
+		}
+
+		metric.Attributes = attributes
+	}
+
+	_, err := client.Metrics.Edit(metric)
+	if err != nil {
+		log.Printf("[INFO] ERROR creating Metric: %s", err)
+		return fmt.Errorf("Error creating Librato metric: %s", err)
+	}
+
+	resource.Retry(1*time.Minute, func() *resource.RetryError {
+		_, _, err := client.Metrics.Get(*metric.Name)
+		if err != nil {
+			if errResp, ok := err.(*librato.ErrorResponse); ok && errResp.Response.StatusCode == 404 {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+
+	return metricReadResult(d, metric)
```

Comment:
Yeah, this approach is used throughout the Librato provider. I'm going to change this for metrics and consider updating the other instances of this pattern as I have time, assuming no side effects.